lib/matplotlib/legend.py

```diff
@@ -565,19 +565,23 @@ def get_legend_handler(legend_handler_map, orig_handle):
         method-resolution-order. If no matching key is found, it
         returns None.
         """
-        legend_handler_keys = list(six.iterkeys(legend_handler_map))
-        if orig_handle in legend_handler_keys:
-            handler = legend_handler_map[orig_handle]
-        else:
+        try:
+            hash(orig_handle)
+        except TypeError:
```

Comment:
This is different behavior, If it is not hash-able, then we need to walk it's MRO to see if we have a handle for any of it's classes.

This is apparently not tested well enough.